logging/src/main/java/org/apache/rocketmq/logging/inner/Logger.java

```diff
@@ -88,8 +88,8 @@ synchronized void closeNestedAppenders() {
         Enumeration enumeration = this.getAllAppenders();
         if (enumeration != null) {
             while (enumeration.hasMoreElements()) {
-                Appender a = (Appender) enumeration.nextElement();
-                if (a instanceof Appender.AppenderPipeline) {
+                if (enumeration instanceof Appender) {
```

Comment:
Why do u think enumeration is an instance of Appender?

Reply:
enumeration  Impossible instanceof Appender.AppenderPipeline，because Appender a = (Appender) enumeration.nextElement();

Reply:
Two kinds of Appender here,  AsyncAppender and WriterAppender. AyscAppender is implementing AppenderPipeline, so it is an instance of AppenderPipeline.

And back to your code, enumeration is definitely not an instance of Appender(enumeration.nextElement() is).

Reply:
thank you，I get it